### PR TITLE
New Feature: getPosFromClick

### PR DESCRIPTION
**Pull request by egidioln**
ctrl+shift+leftClick on pdf view copies to clipboard the cursor coordinates x and y (in cm, w.r.t. bottom left corner of the page)

A detailed explanation of the motivation is given in #2324 and #2307

### CODE REVIEW

File: src/pdfviewer/PDFDocument.cpp [hunk 1]
```diff
@@ -2733,7 +2747,21 @@ void PDFDocument::setupMenus(bool embedded)
...
-
+/*
+ * Copies to clipboard position in cm captured from the pdf viewer (w.r.t. south west corner)
+ */
+void PDFDocument::getPosFromCLick(const QPointF &pos)
+{
+	int page = pdfWidget->pageFromPos(pos.toPoint());
+	if (page < 0) return;
+	float height = (pdfWidget->pageRect(page).height()/pdfWidget->totalScaleFactor()) / 28.45;
+	QClipboard *clipboard = QGuiApplication::clipboard();
```
Comment (sunderme) [inline src/pdfviewer/PDFDocument.cpp:2758]: are you sure about the 28.45 ?
I assume 72/inch = ?

Comment (egidioln) [reply src/pdfviewer/PDFDocument.cpp:2758]: Hey! No, I'm not sure... I found this approximate conversion from pt to cm online and it worked just fine... Sorry I didn't get your suggestion, how is it?


By do way, any suggestions on how to document this feature?

Best!
Lucas

Comment (octaeder) [reply src/pdfviewer/PDFDocument.cpp:2758]: 1pt = 1inch/72.27 = 2.54cm/72.27. So x pt = x / (1/(1pt)) = ( x / (72.27/2.54) )cm where 72.27/2.54 = 28,452756

Comment (octaeder) [reply src/pdfviewer/PDFDocument.cpp:2758]: If pt is postscript point not TeX point then use 72.00 instead of 72.27. Hence result is 28.346457.

Comment (sunderme) [reply src/pdfviewer/PDFDocument.cpp:2758]: poppler says that pageSize is given in points (i.e. 1/72th of an inch)

Comment (egidioln) [reply src/pdfviewer/PDFDocument.cpp:2759]: Thanks! I pushed a new commit with this fix on the conversion. I tested it by picking points from a grid that I drew and the results are way more precise :)

**Comment by sunderme** [comment]
thanks

**Comment by egidioln** [comment]
> In case you want other users to use this, there should be a note in the manual.

Sure, that's also in my todo list... Should I edit https://github.com/texstudio-org/texstudio/blob/master/utilities/manual/usermanual_en.html and open a PR?

**Comment by sunderme** [comment]
yes, as discussed with @octaeder , there is no clear chapter for the pdf viewer yet, so add a section to `3. compile` -> 3.3 pdf viewer -> special features

**Comment by octaeder** [comment]
@egidioln  s. PR #2396, please add your description

**Comment by egidioln** [comment]
> @egidioln s. PR #2396, please add your description

@octaeder  Thanks! I added a paragraph, does it look good?

**Comment by octaeder** [comment]
@egidioln Thanks, I commited your change.

**Comment by octaeder** [comment]
I added your ctrl+shift+click to the magnifier and scroll actions lists

**Pull request merged**